src/devices/dsc.c

```diff
@@ -144,17 +146,25 @@ static int DSC_callback(bitbuffer_t *bitbuffer) {
         esn = (bytes[1] << 16) | (bytes[2] << 8) | bytes[3];
         crc = bytes[4];
 
-        local_time_str(0, time_str);
-
-        if (crc8le(bytes, DSC_CT_MSGLEN, DSC_CT_CRC_POLY, DSC_CT_CRC_INIT) == 0) {
-            printf("%s DSC Contact ESN: %06X, Status: %02X, CRC: %02X\n",
-                time_str, esn, status, crc);
-
-            valid_cnt++; // Have a valid packet.
-        } else if (debug_output) {
-            fprintf(stderr,"%s DSC Contact bad CRC: %06X, Status: %02X, CRC: %02X\n",
-                time_str, esn, status, crc);
+        if (crc8le(bytes, DSC_CT_MSGLEN, DSC_CT_CRC_POLY, DSC_CT_CRC_INIT) != 0) {
+            if (debug_output)
+                fprintf(stderr,"DSC Contact bad CRC: %06X, Status: %02X, CRC: %02X\n",
+                        esn, status, crc);
+            continue;
         }
+
+        sprintf(status_str, "%02x", status);
+        sprintf(esn_str, "%06x", esn);
+        data = data_make(
+                "time", "", DATA_STRING, time_str,
+                "model", "", DATA_STRING, "DSC Contact",
+                "id", "", DATA_STRING, esn_str,
+                "status", "", DATA_STRING, status_str,
```

Comment:
While it is easier to read as hex, status would be easier to process as an int.

Same thing with ID.    The "natural" way for the system to identify devices is with the hex string, but in reality it is a 24 bit unsigned int.   So if I wanted to just read the output the hex string is what I'd want to see.  But for processing it would be easier for it to just be an int.

Reply:
Totally agree. It's a number an should be readable as such. But some devices give the ID in a special format and users expect that. TPMS come to mind, they print the hex ID on the device.
We can get away with just treating it as a "label" for most applications, but the same does not apply for the status -- we really want to parse that. Complicated issue to solve in general way.